keps/sig-scheduling/NNNN-pod-to-node-assignment/README.md

```diff
@@ -0,0 +1,808 @@
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+This Kep proposes adding a feature (`PodToNodeAssignment`) that provides users
+with an explicit API and scheduler support for 1:1 pod to node assignment.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Customers trying to deploy batch workloads on Kubernetes have filed multiple
+requests (See [1](https://github.com/kubernetes/kubernetes/issues/68827),
+[2](https://github.com/kubernetes/kubernetes/issues/105287) for details) for 1:1
+pod-to-node assignment.
+
+Currently there are two approaches to achieve this:
+
+-   set resource requests to something high enough to prevent scheduling any
+    other pods but daemonset pods. This is tricky to do and error prone,
+    especially for smaller nodes.
+-   Use pod anti affinity. Users will need to define the same set of labels on
+    all pods that needs to be repelled, whether or not they actually request 1:1
+    pod to node assignment, it is also expensive to calculate especially for
+    cluster autoscaler.
+
+A simple and native way to provide 1:1 pod-to-node assignment is needed to be
+supported by the scheduler.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+-   Allow users to force 1:1 pod-to-node assignment for a given pod
+
+## Proposal
+
+The idea is to create a scheduler filter plugin such that given a pod and a
+node, a node is filtered out if any of the following is true (assume the API is
+the annotation k8s.io/pod-per-node=true, more on that below):
+
+-   The incoming pod is not a daemonset pod (by checking OwnerReference) and
+    there is a pod with k8s.io/pod-per-node=true already assigned to the node,
+-   If the incoming pod has the k8s.io/pod-per-node=true annotation and the node
+    hosts at least one non daemonset pod. The above logic would also be added to
```

Comment:
both conditions needs to be checked by kubelet, so the sentence "The above logic ..." should be moved to its own bullet.